metaflow/plugins/pypi/conda_environment.py

```diff
@@ -68,14 +68,16 @@ def environments(type_):
             seen = set()
             for step in self.flow:
                 environment = self.get_environment(step)
+                if environment["id_"] is None:
+                    continue
                 if type_ in environment and environment["id_"] not in seen:
                     seen.add(environment["id_"])
                     for platform in environment[type_]["platforms"]:
                         yield environment["id_"], {
                             **{
                                 k: v
                                 for k, v in environment[type_].items()
-                                if k != "platforms"
+                                if k in ["python", "packages"]
```

Comment:
can rethink this change, but was necessary for now as not being explicit about the attributes here meant that we end up passing unsupported ones to the solver, in this case `disabled`